plugin-maven/src/main/java/com/diffplug/spotless/maven/generic/TrimTrailingWhitespace.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2016 DiffPlug
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.diffplug.spotless.maven.generic;
+
+import com.diffplug.spotless.FormatterStep;
+import com.diffplug.spotless.generic.ReplaceRegexStep;
+import com.diffplug.spotless.maven.FormatterStepConfig;
+import com.diffplug.spotless.maven.FormatterStepFactory;
+
+public class TrimTrailingWhitespace implements FormatterStepFactory {
+
+	@Override
+	public FormatterStep newFormatterStep(FormatterStepConfig config) {
+		return ReplaceRegexStep.create("trimTrailingWhitespace", "[ \t]+$", "");
+	}
```

Comment:
Is this how trimTrailingWhitespace is implemented in Gradle?  If so, it should be refactored into lib.  Code duplication makes it hard to maintain later.

Reply:
fixed.

Reply:
fixed